### DISCUSSION TITLE
Airflow DAG Pod stuck in NotReady state

### DISCUSSION THREAD

**Discussion by Ajit-mycom** [apache/airflow#41467] [General]
### Apache Airflow version

Other Airflow 2 version (please specify below)

### If "Other Airflow 2 version" selected, which one?

2.8.1

### What happened?

When I trigger the DAG it created multiple pods but after completion of dag run it's pod stuck in the NotReady state and it remains there in the cluster , I tried alot currently I am using kubernetes executor and I can't change direct to celery as most of the things will impact because of this 

### What you think should happen instead?

There must be any way to do this so once dag run is successfully done it's pod will be deleted .

### How to reproduce

Want some solution or feature so it will delete the pod of dag once dag run is done 

### Operating System

macos 

### Deployment

Official Apache Airflow Helm Chart

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by potiuk** [comment]
I think if you need help from someone (not me I know little about it) - you need to explain a bit more. There is a setting in Airlfow (look it up in airflow k8s executor configuration docs ) a parameter that allows you to not delete POD that completes - disabled by default, pods should be deleted, however maybe you have some other issue. I've seen for example that people's PODs remained undeleted because they had sidecars running. But if you need help from someone here - likely more information and logs would be needed to help them to help you diagnose your problem. 

Converted it into discussion - it does not qualify as reproducible issue.

**Reply by Ajit-mycom** [reply]
Thanks @potiuk for your reply let me share the logs and describe it more

**Reply by Ajit-mycom** [reply]
config:
      webserver:
        expose_config: 'False'
        log_fetch_timeout_sec: 15
      api:
        auth_backends: airflow.api.auth.backend.basic_auth
      kubernetes_executor:
        logs_task_metadata: True
      core:
        remote_logging: '{{- ternary "True" "False" .Values.elasticsearch.enabled }}'
      elasticsearch:
        # host = <host>:<port>
        # write_stdout: True
        json_format: True
      logging:
        colored_console_log: 'True'  
        logging_level: DEBUG
        remote_logging: '{{- ternary "True" "False" .Values.elasticsearch.enabled }}'
    multiNamespaceMode: true
    createUserJob:
      useHelmHooks: false
      applyCustomEnv: false
    migrateDatabaseJob:
      useHelmHooks: false
      applyCustomEnv: false
    useStandardNaming: true
    extraEnv: |-
      - name: AIRFLOW__CORE__FERNET_KEY
        valueFrom:
          secretKeyRef:
            name: dev-3pp-airflow-fernet-key
            key: fernet-key      
    variablesUpdate: true
    rbac:
      create: true    
    executor: "KubernetesExecutor"
    
    this is the config I am using right now and with this it's not deleting the pod which is in NotReady state here what happens is 
    base container is getting terminated with status completed but minio-sync: container keep Running that's why it is not deleting the pod so I want to know if there is any way to kill the sidecar container in airflow with kubernetes executor any setting is there for doing this currently I am removing those manually .
